icu4j/main/core/src/main/java/com/ibm/icu/message2/StringView.java

```diff
@@ -0,0 +1,38 @@
+// © 2024 and later: Unicode, Inc. and others.
+// License & terms of use: https://www.unicode.org/copyright.html
+
+package com.ibm.icu.message2;
+
+class StringView implements CharSequence {
+    final int offset;
+    final String text;
+
+    StringView(String text, int offset) {
+        this.offset = offset;
+        this.text = text;
+    }
+
+    StringView(String text) {
+        this(text, 0);
+    }
+
+    @Override
+    public int length() {
+        return text.length() - offset;
+    }
+
+    @Override
+    public char charAt(int index) {
```

Comment:
Because that is the interface I am implementing.
`StringView implements CharSequence`
And [`CharSequence`](https://docs.oracle.com/javase/8/docs/api/java/lang/CharSequence.html) asks for `char charAt(int index)`